bookkeeper-server/pom.xml

```diff
@@ -178,6 +178,11 @@
       <artifactId>http-server</artifactId>
       <version>${project.version}</version>
     </dependency>
+    <dependency>
+      <groupId>com.carrotsearch</groupId>
```

Comment:
Do you have checked license for this?
Is it really needed? We already have several custom collections, maybe we could have our own implementation of IntArrayList without importing a whole third party lib. This will be used on client side.
It is used in EnsemblePlacement policy which is very often customized, at least in my projects.
I would not like to impose such dep on public plugins API

Reply:
second to this comment here:

- for license, we need to update src/main/resources/NOTICE.bin.txt
- it is a new dependency introduced on client. It would prefer if we can avoid using this dependency.
- I am not sure what would be the benefits of using IntArrayList. If boxing, unboxing Integer and using an ArrayList is a problem, why can't we use `int[]` because we don't need resizing ability? we can wrap `int[]` into a recycle class, it should be simple and efficient, no?

Reply:
> Do you have checked license for this? 
License is Apache v2.

> I am not sure what would be the benefits of using IntArrayList. If boxing, unboxing Integer and using an ArrayList is a problem, why can't we use int[] because we don't need resizing ability? we can wrap int[] into a recycle class, it should be simple and efficient, no? 

Yes, basically the class is a wrapper on top of an `int[]`. I guess the option here would be to either reimplement the "set" logic on the `int[]` or maybe import just the needed class. 

> We already have several custom collections, maybe we could have our own implementation of IntArrayList without importing a whole third party lib

Yes, the other collections were more about having no-boxing and concurrent collections. For non-concurrent primitive collections there are many available implementations (like the HPPC here).